api/desecapi/serializers.py

```diff
@@ -775,14 +778,24 @@ class Meta:
 
 
 class AuthenticatedActivateUserActionSerializer(AuthenticatedBasicUserActionSerializer):
+    captcha = CaptchaSolutionSerializer(required=False)
 
     class Meta(AuthenticatedBasicUserActionSerializer.Meta):
         model = models.AuthenticatedActivateUserAction
-        fields = AuthenticatedBasicUserActionSerializer.Meta.fields + ('domain',)
+        fields = AuthenticatedBasicUserActionSerializer.Meta.fields + ('captcha', 'domain',)
         extra_kwargs = {
             'domain': {'default': None, 'allow_null': True}
         }
 
+    def to_internal_value(self, data):
+        internal_value = super().to_internal_value(data)
+        try:
+            internal_value.pop('captcha')  # remove captcha from internal value to avoid passing to Meta.model(**kwargs)
+        except KeyError:
+            if internal_value['user'].needs_captcha:
+                raise serializers.ValidationError({'captcha': fields.Field.default_error_messages['required']})
```

Comment:
can this be raised during the validation phase?

Reply:
Fixed. I was somehow under the impression that `validate()` runs before `to_internal_value()` so that the user object would not yet be instantiated, but in fact it's the other way around.